code/addons/docs/template/stories/docspage/toc.stories.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+
+import { styled } from 'storybook/internal/theming';
+
+import type { Meta, StoryObj } from '@storybook/react-vite';
+
+import { expect, userEvent, within } from 'storybook/test';
+
+import { Heading } from '../../../src/blocks/blocks/Heading';
```

Comment:
Looks like this line is breaking CI @Sidnioulz

Reply:
I've tried a few alternatives, none working in the CI. Who could I ask help from to get this through? Writing the stories was instrumental in me correctly handling all the edge cases, so I'd love to preserve that stories file or have an equivalent UT file.

Reply:
The issue is that you have put the toc.stories.tsx into the `template` folder, which gets copied over to each sandbox. That means, that you shouldn't use any framework specific imports (like type imports from `@storybook/react-vite`, and second, the TableOfContents component requires a DocsRenderer Provider being in place, which isn't set up by the sandboxes. Only the Storybook UI's Storybook is setting up the necessary DocsRenderer Provider, so that we can actually render the components inside of the Storybook preview.

I have pushed a commit to move the story outside of the `templates` folder. Now, it is tested only by Storybook UI's Storybook.

Reply:
Thanks for the detailed explanation! I did not know about the role of the `templates` folder.